### ISSUE TITLE
File with character '#' in filename can't be opened

### ISSUE THREAD

**Issue by d10r** [nextcloud/richdocuments#161]
### Steps to reproduce
Create a file with '#' in it's name, e.g. "test#1.odt".
Try to open it (click it).

Collabora launches, but only shows the error message:
> Failed to load the document. Please ensure the file type is supported and not corrupted, and try again.

The collabora log shows:
```
kit-00050-00056 09:21:07.336737 [ lokit_004 ] ERR  Failed to load: file:///user/docs/fHrC2B0NDca13Xsa/test#1.odt, error: Unsupported URL <file:///user/docs/fHrC2B0NDca13Xsa/test#1.odt>: "type detection failed"| kit/Kit.cpp:1484
kit-00050-00056 09:21:07.336800 [ lokit_004 ] ERR  Failed to get LoKitDocument instance.| kit/ChildSession.cpp:363
```

After renaming the file to `test1.odt`, it can be opened.

### Server configuration
**Operating system**: Ubuntu 16.04
**Web server:** Apache 2.4
**Database:** mariadb-server 10.0
**PHP version:** 7.0
**Nextcloud version:** 12.0.4

`https://cloud.lab10.io/index.php/settings/integrity/failed` returns:
`No errors have been found.`

**Comment by oparoz** [comment]
Confirmed

**Comment by thebearon** [comment]
This is fixed in Collabora Online 3.1.
